src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2209,6 +2271,12 @@ private void onDecodeData(
                     }
                 }
             }
+            else
+            {
+                progress += payloadLength.value;
+            }
```

Comment:
Why do we need this `else` when `exchange` is `null`?
`progress` should not advance in that case, agreed?

Reply:
What I was thinking is when we get reset while decoding is still in progress I thought we should progress till the end of the data frame. What do you think?

Reply:
Thinking about when it would be appropriate to receive an `H2_DATA` frame for a `streamId` that has no `exchange`...
Perhaps a race where we have sent `RST_STREAM` back to the client but the inbound `H2_DATA` is already in flight and arrives afterwards.
Let's keep this behavior, but flip the condition logic and put the `if (exchange == null)` case first to make it more clearly deliberate.